Copy nested fragments

louy opened the pull request:
Closes #98.

slindberg commented on packages/model-fragments/lib/fragments/model.js:
```diff
@@ -170,13 +170,21 @@ var ModelFragment = CoreModel.extend(Ember.Comparable, Ember.Copyable, {
+      } else {
+        newFragment.set(attribute.name, this.get(attribute.name));
+      }
+    }.bind(this));
```
`forEach` accepts a context object as its second argument, I'd prefer that you use that instead.

slindberg commented on packages/model-fragments/lib/fragments/model.js:
```diff
@@ -170,13 +170,21 @@ var ModelFragment = CoreModel.extend(Ember.Comparable, Ember.Copyable, {
+    get(this.constructor, 'attributes').forEach(function(attribute) {
+      if( attribute.type === 'fragment' ) {
+        if( attribute.kind === 'hasMany' ) {
+          newFragment.set(attribute.name, this.get(attribute.name).map(Ember.copy));
```
Can you use the `get` var here and two places below (like you did on line 175).

slindberg commented on packages/model-fragments/lib/fragments/model.js:
```diff
@@ -170,13 +170,21 @@ var ModelFragment = CoreModel.extend(Ember.Comparable, Ember.Copyable, {
+    
+    var newFragment = store.createFragment(type);
+    get(this.constructor, 'attributes').forEach(function(attribute) {
+      if( attribute.type === 'fragment' ) {
```
Small nit: can you make the spaces around parens consistent with the rest of the repo? e.g. `if (attribute.type === 'fragment') {`

slindberg commented:
I really like your approach to this! Could you address my inline comments and rebase? I'd also really really appreciate if you added a test to [packages/model-fragments/tests/integration/fragments/nested_test.js](https://github.com/lytics/ember-data.model-fragments/blob/v0.4.0/packages/model-fragments/tests/integration/fragments/nested_test.js) 😄

louy commented:
Could you please do those changes for me? Unfortunately, I deleted my local fork and I'm now unable to access these commits. The Ember project I was working on is done now and that's probably why I removed my fork. I forgot that I had this PR open. Sorry about that.

louy commented:
Just found out that it's possible to checkout a PR. Closing in favor of #123.
